### PR TITLE
Display ReportActionContextMenu

### PR DESCRIPTION
**Pull request by roryabraham**
...bear with me while I gather screenshots...

### Details
Displays the ReportActionContextMenu, hidden behind a beta. This merges the feature branch we've been building this feature on into master (using a beta on master instead of a feature branch will be less of a hassle going forward).

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/147472

### Tests
1. Run the app.
2. Test that the `ReportActionContextMenu` displays correctly.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web
image

image

#### Desktop
image

image

(plus made sure that there were no regressions to the CreateMenu popover:

image

### CODE REVIEW

**Comment by roryabraham** [comment]
Sorry for the noise, I've got to come back to this one tonight. Throwing a WIP up in the meantime. Will re-ping when ready for review.

**Comment by roryabraham** [comment]
Okay, this is no longer WIP, ready for review! Fixed a bug on iOS and retested all platforms 👍

**Review by marcaaron — DISMISSED** [review]
LGTM. I'm going to approve this, but it would be good if we can clean up the propTypes stuff at some point. I just had one other comment about documenting the purpose of `PopoverWithMeasuredContent`. After not looking at this for a while I can't remember at all why we need that. So it's probably worth documenting in the code.

File: src/pages/home/report/ReportActionItem.js
Comment (roryabraham) [reply src/pages/home/report/ReportActionItem.js:137]: > Why are we not just providing a constant width / height?

Mostly just because we want the `ReportActionContextMenu` to be responsively sized, and adding or removing elements from the menu is as simple as adding or removing items from the `CONTEXT_ACTIONS` array.

_However_, one potential simplification would be to eliminate the `measureContent` component and just measure the children (and pass a `ReportActionContextMenu` that's always visible in the children).

**Comment by marcaaron** [comment]
Oh yeah also I am still not sold on this animation I think I've mentioned it before but the scaling feels busy (hard to tell from looking at this video).

https://user-images.githubusercontent.com/32969087/110856337-9ba74e80-825b-11eb-8067-249bbac8cd4d.mp4

File: src/components/Hoverable/HoverablePropTypes.js
Comment (roryabraham) [reply src/components/Hoverable/HoverablePropTypes.js:7]: Good call, this doesn't make a ton of sense.

**Comment by roryabraham** [comment]
> Oh yeah also I am still not sold on this animation I think I've mentioned it before but the scaling feels busy

Yeah, I'd prefer to tackle this in a follow-up. One solution that I think would solve the busyness problem would be to give the animation a 500ms delay (so the tooltip only displays after an element has been hovered for half a second). That's pretty common with tooltips too I think.

**Comment by roryabraham** [comment]
@marcaaron Added a comment for the `PopoverWithMeasuredContent` and nixed the required type in `oneOfType`. Since the `_.omit` propTypes was NAB I didn't do that for now (although you've convinced me that it's a bit of an anti-pattern that should probably be avoided in the future. Can do that in a mini follow-up.

**Review by marcaaron — APPROVED** [review]
LGTM

**Review by jasperhuangg — APPROVED** [review]

**Review by shawnborton — APPROVED** [review]

**Comment by shawnborton** [comment]
All you @Dal-Papa

**Comment by Dal-Papa** [comment]
I'm out of my depth on this PR and it seems that everybody came to a good conclusion so I'm removing myself.

**Comment by roryabraham** [comment]
Okay, I don't want any merge conflicts so I'm going to go ahead and self-merge since I've got 3 approvals!

**Pull request merged**
